pandas/core/base.py

```diff
@@ -207,11 +207,14 @@ def _obj_with_exclusions(self):
         if self._selection is not None and isinstance(self.obj, ABCDataFrame):
             return self.obj.reindex(columns=self._selection_list)
 
-        if len(self.exclusions) > 0:
-            return self.obj.drop(self.exclusions, axis=1)
-        else:
+        if not isinstance(self.obj, ABCDataFrame) or not self.exclusions:
```

Comment:
use elif and reverse the conditions, e.g. not self.exclusios and not isinstance.....